docroot/modules/custom/webform_custom_submissions/src/TravelformHandler.php

```diff
@@ -0,0 +1,486 @@
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Psr7\Response;
+use \Exception;
+
+require_once('../travel_services_config.inc');
+
+class TravelformHandler {
+
+  protected $create_issue_url;
+  protected $domestic_project;
+  protected $international_project;
+  protected $submission_client;
+  protected $username;
+  protected $vouchers_project;
+  protected $field_helper;
+
+  public function __construct() {
+    global $CREATE_ISSUE_URL;
+    global $DOMESTIC_PROJECT;
+    global $INTERNATIONAL_PROJECT;
+    global $USERNAME;
+    global $VOUCHERS_PROJECT;
+
+    $this->create_issue_url = $CREATE_ISSUE_URL;
+    $this->domestic_project = $DOMESTIC_PROJECT;
+    $this->international_project = $INTERNATIONAL_PROJECT;
+    $this->submission_client = new Client(['base_uri' => $CREATE_ISSUE_URL]);
+    $this->username = explode(':', $USERNAME);
+    $this->vouchers_project = $VOUCHERS_PROJECT;
+    $this->field_helper = new FieldHelper();
+  }
+
+  /*
+   * Script for getting, formatting and submitting travel services form data to JIRA
+   */
+
+  public function submitToJira($form, &$form_state) {
+    try {
+      $postData = $this->compilePOSTData();
+      $postResponse = $this->sendPOSTData($postData);
+
+      //Uncomment to show full repsonse in broswer
+      //drupal_set_message(json_decode(var_export($postResponse, true)));
+      //var_export($_POST,true);
+
+      //dpm($_POST);
+      //drupal_set_message($postData, 'error');
+      //drupal_set_message($postResponse, 'error');
+      $decoded = json_decode($postResponse, TRUE);
+      if (isset($decoded['id'])) {
+        $page = explode('/', \Drupal::service('path.current')->getPath());
+        \Drupal::logger('Travel Services Response ' . $page[sizeof($page) - 1])->notice($postResponse);
+        $issueId = $this->getIssueId($postResponse);
+
+        $filesUploaded = $this->attachFiles($issueId, $form_state);
+        $this->parseIssueResponse($postResponse, $filesUploaded, $_POST['formTitle'], $postData, $_FILES);
+      } elseif (isset($decoded['errorMessages'])) {
+        \Drupal::logger('Travel Services Error')->notice($postResponse);
+        drupal_set_message(t('There was an error processing your request. Code-0001'), 'error');
+        mail('zerihun.tegegn@cgifederal.com', 'TravelServices Error', $postResponse);
+      } else {
+        drupal_set_message(t('There was an error processing your request. Code-0002'), 'error');
+        \Drupal::logger('Travel Services Error')->notice('Unidentified Error: JIRA Response = ' . $postResponse);
+      }
+    } catch (Exception $e) {
+      \Drupal::logger('Travel Services Exception')->notice($e->getMessage());
+      drupal_set_message(t('Unable to process request at this time, please try again later.'), 'error');
+      //drupal_set_message($e->getMessage());
+    }
+  }//end submitToJira
+
+//Compiles POST data and returns the data formatted as JSON
+  protected function compilePOSTData() {
+
+    $dropDowns = array(
```

Comment:
We'll plan time next week once the actual service is squared away to work with @jdearie  and mappings from current forms.